.github/workflows/release.yml

```diff
@@ -0,0 +1,32 @@
+name: Create a draft release
+
+on:
+  push:
+    branches: [main]
+    tags:
+      - "v*" # Push events to matching v*, i.e. v1.0, v20.15.10
+
+jobs:
+  autorelease:
+    name: Create release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+      - name: Release Notes
+        run: |
+          git log $(git describe HEAD~ --tags --abbrev=0)..HEAD --pretty='format:* %h %s%n' --no-merges >> ".github/release_template.md"
```

Comment:
since we maintain [a changelog](https://github.com/microsoft/kiota-authentication-azure-python/blob/main/CHANGELOG.md) I wonder why not take the release notes from there?

If you want, here is a script you can tweak
https://github.com/microsoft/kiota/blob/main/scripts/get-release-notes.ps1

Alternatively there are many github actions that can read from the keepachangelog format